index.bs

```
@@ -409,8 +412,51 @@ dictionary MediaMetadataInit {
     </li>
     <li>
       Set <var>metadata</var>'s {{MediaMetadata/artwork}} using
-      <var>init</var>'s {{MediaMetadataInit/artwork}} by calling the
-      <a><code>MediaImage(init)</code></a> constructor.
+      <var>init</var>'s {{MediaMetadataInit/artwork}} by performing the
+      following steps:
+      <ol>
+        <li>
+          Let <var>frozenArtwork</var> be an empty list of type {{MediaImage}}.
+        </li>
+        <li>
+          For each <var>entry</var> in <var>init</var>'s
+          {{MediaMetadataInit/artwork}}, perform the following steps:
+          <ol>
+            <li>
+              Let <var>image</var> be a new {{MediaImage}}.
+            </li>
+            <li>
+              Parse <var>entry</var>'s {{MediaImage/src}} into an <a>absolute
+              URL</a> using the <a>document base URL</a>, set <var>image</var>'s
+              {{MediaImage/src}} to the return value.
+            </li>
+            <li>
+              Set <var>image</var>'s {{MediaImage/sizes}} to <var>entry</var>'s
+              {{MediaImage/sizes}}.
+            </li>
+            <li>
+              Set <var>image</var>'s {{MediaImage/type}} to <var>entry</var>'s
+              {{MediaImage/type}}.
+            </li>
+            <!-- XXX IDL dictionaries are usually returned by value, so don't
+            need to be immutable. But FrozenArray reifies the dictionaries to
+            mutable JS objects accessed by reference, so we explicitly freeze
+            them. It would be better to do this with IDL primitives instead of
+            JS - see https://www.w3.org/Bugs/Public/show_bug.cgi?id=29004 -->
+            <li>
+              Call <a lt="freeze">Object.freeze</a> on <var>image</var>, to
+              prevent accidental mutation by scripts.
```

I don't know if we need to do this. My understanding from reading the bug is that freezing the dictionary was needed because `[SameObject]` was such that changing the value would be a problem. I guess in our guess it wouldn't because we can't use `[SameObject]`, making accidental change by script harmless. I guess if we were to add this, it would throw an exception instead of only updating the copy of the returned array.

@annevk, do you have an opinion about this?